### ISSUE TITLE
[Task]: Serve settings for module in the context of creator

### ISSUE THREAD

**Issue by subrata71** [appsmithorg/appsmith#28859]
### Is there an existing issue for this?

- [X] I have searched the existing issues

TL;DR
settings for module in the context of creator = {plugin specific settings for the public entity} minus {module instance settings}

Generate settings in the context of the creator by removing the settings available in the context of the consumer.

### Impacted endpoint

GET - Package details by packageId (Used by Package Editor)
Endpoint (EDIT MODE)- api/v1/packages/{packageId}

### Example
Settings that are overridable by the consumer are the following

```
{
  "setting": [
    {
      "sectionName": "",
      "id": 1,
      "children": [
        {
          "label": "Run query on page load",
          "configProperty": "executeOnLoad",
          "controlType": "SWITCH",
          "subtitle": "Will refresh data each time the page is loaded"
        },
        {
          "label": "Request confirmation before running query",
          "configProperty": "confirmBeforeExecute",
          "controlType": "SWITCH",
          "subtitle": "Ask confirmation from the user each time before refreshing data"
        }
      ]
    }
  ]
}
```

Settings that are overridable in regular action for a given plugin e.g. Postgres are the following

```
[
  {
    "sectionName": "",
    "id": 1,
    "children": [
      {
        "label": "Run query on page load",
        "configProperty": "executeOnLoad",
        "controlType": "SWITCH",
        "subtitle": "Will refresh data each time the page is loaded"
      },
      {
        "label": "Request confirmation before running query",
        "configProperty": "confirmBeforeExecute",
        "controlType": "SWITCH",
        "subtitle": "Ask confirmation from the user each time before refreshing data"
      },
      {
        "label": "Use Prepared Statement",
        "subtitle": "Turning on Prepared Statement makes your queries resilient against bad things like SQL injections. However, it cannot be used if your dynamic binding contains any SQL keywords like 'SELECT', 'WHERE', 'AND', etc.",
        "configProperty": "actionConfiguration.pluginSpecifiedTemplates[0].value",
        "controlType": "SWITCH",
        "initialValue": true
      },
      {
        "label": "Query timeout (in milliseconds)",
        "subtitle": "Maximum time after which the query will return",
        "configProperty": "actionConfiguration.timeoutInMillisecond",
        "controlType": "INPUT_TEXT",
        "dataType": "NUMBER"
      }
    ]
  }
]
```

### Outcome

The output should consist of those settings that are not present in the settings of module consumer.

```
[
  {
    "sectionName": "",
    "id": 1,
    "children": [
      {
        "label": "Use Prepared Statement",
        "subtitle": "Turning on Prepared Statement makes your queries resilient against bad things like SQL injections. However, it cannot be used if your dynamic binding contains any SQL keywords like 'SELECT', 'WHERE', 'AND', etc.",
        "configProperty": "actionConfiguration.pluginSpecifiedTemplates[0].value",
        "controlType": "SWITCH",
        "initialValue": true
      },
      {
        "label": "Query timeout (in milliseconds)",
        "subtitle": "Maximum time after which the query will return",
        "configProperty": "actionConfiguration.timeoutInMillisecond",
        "controlType": "INPUT_TEXT",
        "dataType": "NUMBER"
      }
    ]
  }
]
```

**Comment by ashit-rath** [comment]
@subrata71 double checking for my understanding

Creator -> `api/v1/packages/{packageId}` - will return modules with settingsForm only available for creator side
Consumer -> `api/v1/packages?workspaceId=<WORKSPACE_ID>` - will return modules with settingsForm only available for consumer side

**Comment by subrata71** [comment]
That's correct. Does the output given in the description make sense to you?
